lib/LoginStrategy.ts

```diff
@@ -422,4 +401,41 @@ export class LoginStrategy extends BaseStrategy {
 
     return true;
   }
+
+  /**
+   * Filters verified roles into valid and invalid roles and sets validation status
```

Comment:
I beleive that `invalid` isn't appropriate in this case. May be `notSaticfiedRoles`. Also `valid` isn't needed, because user most often need to know which requirements he yet need to satisfy

Reply:
Valid credentials are needed, since we display the verified roles once authorised in `ram-client-examples`. PDA has always returned verified roles.

Reply:
Ok, thanks. Just to clarify, is it assumed that `validate` returns only those user roles which present in accepted?

Reply:
It depends on another parameter `includeAllRoles` - https://github.com/energywebfoundation/passport-did-auth/blob/6028f44f026b615e93db5095b4c530c7d1d5c1b4/lib/LoginStrategy.ts#L66.

So, if this is true it will return all verified roles and if not then only the accepted roles.

Reply:
But isn't `filterUserRoleStatus` rerurns only accepted roles regardless of `includeAllRoles`?

Reply:
I think we just shouldn't call `filterUsrRoleStatus`  when `includeAllRoles`

Reply:
@JGiter Thinking more on the above, what should we do of revoked or expired roles if the `includeAllRoles` is true? Should it be separated in a different list?

Reply:
@JGiter @nichonien where is `filterUserRoleStatus`? I don't see it in this code

> Thinking more on the above, what should we do of revoked or expired roles if the includeAllRoles is true? Should it be separated in a different list?

@nichonien does your question about still stand? To me, they can just be roles in the `userRoles` array that are marked as `revoked` or `expired`

Reply:
> @JGiter @nichonien where is filterUserRoleStatus? I don't see it in this code

I renamed it since we decided to return just one array of roles.